packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -1,366 +1,100 @@
-import type { AssetId } from '@shapeshiftoss/caip'
-import { CHAIN_NAMESPACE, fromAssetId, solAssetId } from '@shapeshiftoss/caip'
-import type { GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
-import type { KnownChainIds } from '@shapeshiftoss/types'
-import type { Result } from '@sniptt/monads'
 import { Err, Ok } from '@sniptt/monads'
 import type { AxiosError } from 'axios'
-import { v4 as uuid } from 'uuid'
 
-import { getDefaultSlippageDecimalPercentageForSwapper } from '../../../constants'
 import type {
   CommonTradeQuoteInput,
-  GetUtxoTradeQuoteInput,
-  SwapErrorRight,
+  GetTradeRateInput,
   SwapperDeps,
-  TradeQuote,
+  TradeQuoteResult,
 } from '../../../types'
-import {
-  type GetEvmTradeQuoteInput,
-  type ProtocolFee,
-  SwapperName,
-  TradeQuoteError,
-} from '../../../types'
-import { getRate, makeSwapErrorRight } from '../../../utils'
 import {
   CHAINFLIP_BAAS_COMMISSION,
-  CHAINFLIP_BOOST_SWAP_SOURCE,
   CHAINFLIP_DCA_BOOST_SWAP_SOURCE,
-  CHAINFLIP_DCA_QUOTE,
   CHAINFLIP_DCA_SWAP_SOURCE,
-  CHAINFLIP_REGULAR_QUOTE,
-  CHAINFLIP_SWAP_SOURCE,
-  usdcAsset,
 } from '../constants'
-import type { ChainflipBaasQuoteQuote, ChainflipBaasQuoteQuoteFee } from '../models'
-import { chainflipService } from '../utils/chainflipService'
-import { getEvmTxFees } from '../utils/getEvmTxFees'
-import { getUtxoTxFees } from '../utils/getUtxoTxFees'
-import { getChainFlipIdFromAssetId, isSupportedAssetId, isSupportedChainId } from '../utils/helpers'
-
-export const _getTradeQuote = async (
+import {
+  calculateChainflipMinPrice,
+  getChainFlipIdFromAssetId,
+  getChainFlipSwap,
+} from '../utils/helpers'
+import { _getTradeRate } from './getTradeRate'
+
+// This isn't a mistake. A trade quote *is* a trade rate. Chainflip doesn't really have a notion of a trade rate,
+// they do have a notion of a "swap" (which we effectively only use to get the deposit address), which is irrelevant to the notion of quote vs. rate
+export const getTradeQuote = async (
   input: CommonTradeQuoteInput,
   deps: SwapperDeps,
-): Promise<Result<TradeQuote[], SwapErrorRight>> => {
-  const {
-    sellAsset,
-    buyAsset,
-    accountNumber,
-    receiveAddress,
-    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
-    affiliateBps: commissionBps,
-  } = input
+): Promise<TradeQuoteResult> => {
+  const maybeTradeRates = await _getTradeRate(input as unknown as GetTradeRateInput, deps)
```

Comment:
Same here, we probably don't want to lie to the compiler or things may break in dangerous ways. This is a quote input, not a rate input, and the `_getTradeRate` method will (at least according to types) return undefined `accountNumber` and `receiveAddress`, which is not a valid quote so we want to ensure:
1. input is a trade quote input (no dangerous casting)
2. output is a trade quote (fine to reuse core logic, but according to types, this outputs a trade rate which we cannot safely reuse)

Reply:
Refactor with traderate extraction